serialization/untyped_boolean.go

```diff
@@ -0,0 +1,23 @@
+package serialization
+
+import "errors"
+
+// UntypedBoolean defines an untyped boolean object.
+type UntypedBoolean struct {
+	UntypedNode
+}
+
+// GetValue returns the bool value.
+func (un *UntypedBoolean) GetValue() *bool {
+	return un.value.(*bool)
```

Comment:
this is going to cause a panic
```suggestion
    castValue, ok := un.value.(*bool)
    if ok {
	   return castValue
	}
	return nil
```